src/main/java/org/folio/rest/impl/CheckinReceivePiecesHelper.java

```diff
@@ -737,4 +726,19 @@ private boolean isMissingLocation(PoLine poLine, Piece piece) {
       }
     });
   }
+
+  CompletionStage<List<CompletableFuture<Void>>> removeForbiddenEntities(List<PoLine> poLines, Map<String, Map<String, T>> pieces) {
+    Map<String, List<PoLine>> poLinesGroupedByOrderId = poLines.stream().collect(groupingBy(PoLine::getPurchaseOrderId));
+    String query = buildQuery(convertIdsToCqlQuery(new ArrayList<>(poLinesGroupedByOrderId.keySet())), logger);
```

Comment:
To avoid creation of new list you can just change signature of the `convertIdsToCqlQuery` to accept `Collection<String>`